### PR TITLE
[SHELL32][SDK] Implement SHIsTempDisplayMode

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Implementing missing features...
JIRA issue: [CORE-9277](https://jira.reactos.org/browse/CORE-9277)

## Proposed changes

- Implement `SHIsTempDisplayMode` function.
- Add `SHIsTempDisplayMode` prototype to `<undocshell.h>`.

### CODE REVIEW

**Comment by HBelusca** [comment]
(It would be more helpful to implement those functions that are known to be used by other modules of the shell. Do you know which module(s) in Windows use that function?)

**Comment by katahiromz** [comment]
image
`CDesktopBrowser` and `CActiveDesktop` are using `SHIsTempDisplayMode`.

File: dll/win32/shell32/stubs.cpp [hunk 1]
```diff
@@ -141,15 +141,34 @@ GetFileDescriptor(FILEGROUPDESCRIPTOR *pFileGroupDesc, BOOL bUnicode, INT iIndex
...
- * Unimplemented
+/*************************************************************************
+ *  SHIsTempDisplayMode [SHELL32.724]
+ *
+ * Is the current display settings temporary?
  */
-EXTERN_C BOOL
-WINAPI
-SHIsTempDisplayMode(VOID)
+EXTERN_C BOOL WINAPI SHIsTempDisplayMode(VOID)
```
[the remark this answers is not in the document]
Comment (katahiromz) [reply dll/win32/shell32/stubs.cpp]: Moved to `CDesktopBrowser.cpp`.

**Comment by katahiromz** [comment]
> Do you know which module(s) in Windows use that function?

At least, `CDesktopBrowser` and `CActiveDesktop` use this function.

**Review by learn-more — APPROVED** [review]

**Comment by SergeGautherie** [comment]
No code nor comment about number of colors?

While here, shouldn't we record somewhere/somehow that this function is supposed to be Vista+?

**Comment by katahiromz** [comment]
`shell32.spec`:
```txt
724 stdcall -noname SHIsTempDisplayMode()
```

`SHIsTempDisplayMode` has no name.

**Pull request merged**